errors: Introduce internal package

efritz opened the pull request.

camdencheek reviewed (approved):
I'm going to trust that all the imports are right. The linting rules and the new package LGTM. 

Could you elaborate on the motivation here? Nominally, it's "behavior clashes like multierror swallowing errors" but I don't actually know why multierror would do that.

efritz commented:
@camdencheek I updated the PR description with a bit more context. I actually linked a relevant issue (but not the one I described in the text). Turns out I only complained about this in our #lang-go slack channel and there wasn't a backing issue for it.

I've also split up the changes into multiple commits to bucket the hand changes from the mechanical changes. I've also hand-vetted all 830 files to make sure only imports changed (or the code I had to change to hand to fix name clashes).

![Screen Shot 2022-02-04 at 1 26 19 PM](https://user-images.githubusercontent.com/103087/152591925-8fa0ef40-d239-4c2c-aeb3-38a2d7938733.png)

(Wasn't quick 😰 )

camdencheek commented:
Oof, yeah, that's no good. That made me go check that all my assumptions about how `errors.Is` and `errors.As` work are true when used with wrapping and multierror. Turns out they are, and that seems to just be a formatting problem, but making wrapped errors invisible is no bueno.

bobheadxi reviewed (approved):
nice, thanks for cleaning this up!

unknwon reviewed (approved):
<img src="https://media1.giphy.com/media/SY2hQpAMLnuxtgLT5C/giphy.gif"/>

mrnugget commented:
> I've also hand-vetted all 830 files to make sure only imports changed

Iiiiinnterrrrreeesting... Gonna share this in our private `#is-eric-fritz-actually-multiple-people` channel.
